src/coreComponents/meshUtilities/CornerPointMeshGenerator.cpp

```diff
@@ -0,0 +1,191 @@
+  MeshBody & meshBody = meshBodies.registerGroup< MeshBody >( this->getName() );
+
+  MeshLevel & meshLevel0 = meshBody.registerGroup< MeshLevel >( string( "Level0" ));
+  NodeManager & nodeManager = meshLevel0.getNodeManager();
+  CellBlockManager & cellBlockManager = domain.getGroup< CellBlockManager >( keys::cellManager );
+
+  // Step 0: transfer the neighbor list
+
+  // TODO: change the name, we don't use metis
+  domain.getMetisNeighborList() = m_cPMeshBuilder->neighborsList();
+
+  // we can start constructing the mesh
+
+  // Step 1: fill vertex information
+
+  arrayView2d< real64 const > vertices = m_cPMeshBuilder->vertices();
+  arrayView1d< globalIndex const > vertexToGlobalVertex = m_cPMeshBuilder->vertexToGlobalVertex();
+  localIndex const nVertices = vertices.size( 0 );
+  nodeManager.resize( nVertices );
+  arrayView1d< globalIndex > const & vertexLocalToGlobal = nodeManager.localToGlobalMap();
+  arrayView2d< real64, nodes::REFERENCE_POSITION_USD > const & X = nodeManager.referencePosition();
+
+  Group & vertexSets = nodeManager.sets();
+  SortedArray< localIndex > & allVertices =
+    vertexSets.registerWrapper< SortedArray< localIndex > >( string( "all" ) ).reference();
+
+  real64 xMin[3] = { std::numeric_limits< real64 >::max() };
+  real64 xMax[3] = { std::numeric_limits< real64 >::min() };
+
+  for( localIndex iVertex = 0; iVertex < nVertices; ++iVertex )
+  {
+    X( iVertex, 0 ) = vertices( iVertex, 0 );
+    X( iVertex, 1 ) = vertices( iVertex, 1 );
+    X( iVertex, 2 ) = vertices( iVertex, 2 );
+    allVertices.insert( iVertex );
+    vertexLocalToGlobal( iVertex ) = vertexToGlobalVertex( iVertex );
+
+    for( int dim = 0; dim < 3; dim++ )
+    {
+      if( X( iVertex, dim ) > xMax[dim] )
+      {
+        xMax[dim] = X( iVertex, dim );
+      }
+      if( X( iVertex, dim ) < xMin[dim] )
+      {
+        xMin[dim] = X( iVertex, dim );
+      }
+    }
+  }
+  LvArray::tensorOps::subtract< 3 >( xMax, xMin );
+  meshBody.setGlobalLengthScale( LvArray::tensorOps::l2Norm< 3 >( xMax ) );
+
+  // Step 2: fill cell information
+
+  CellBlock * cellBlock = &cellBlockManager.getGroup( keys::cellBlocks ).registerGroup< CellBlock >( "DEFAULT_HEX" );
+
+  // temporary accessors while we only support the conforming case
+  // this is not what we ultimately want, which is instead:
+  //  -> map from elem to faces
+  //  -> map from face to nodes
+  // what is below is a temporary mess (that maybe, should be hidden in CPMeshData)
+  arrayView1d< localIndex const > activeCellInsidePartitionToActiveCell = m_cPMeshBuilder->activeCellInsidePartitionToActiveCell();
+  arrayView1d< globalIndex const > activeCellToGlobalCell = m_cPMeshBuilder->activeCellToGlobalCell();
+  arrayView1d< localIndex const > activeCellToCell = m_cPMeshBuilder->activeCellToCell();
+  arrayView1d< localIndex const > cellToCPVertices = m_cPMeshBuilder->cellToCPVertices();
+  arrayView1d< localIndex const > cPVertexToVertex = m_cPMeshBuilder->cPVertexToVertex();
```

Comment:
Instead of exposing the vectors like that, maybe just having a member function `m_cPMeshBuilder::activeCellToCell` would make the notations lighter?